src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -56,10 +57,22 @@ public IEnumerable<LockFileTargetLibrary> GetRuntimeLibraries(IEnumerable<string
                 allExclusionList.UnionWith(privateAssetsExclusionList);
             }
 
-            if(_filterlockFileTarget != null)
+            if(_packagesToBeFiltered != null)
             {
-                IEnumerable<LockFileTargetLibrary> filterLibraries = _filterlockFileTarget.Libraries;
-                Dictionary<string, LockFileTargetLibrary> filterLookup = filterLibraries.ToDictionary(e => e.Name, StringComparer.OrdinalIgnoreCase);
+                var filterLookup = new Dictionary<string, HashSet<PackageInfo>>();
+                foreach (var pkg in _packagesToBeFiltered)
+                {
+                    if (filterLookup.ContainsKey(pkg.Name))
+                    {
+                        filterLookup[pkg.Name].Add(pkg);
```

Comment:
The key is case-sensitive , as you can see from definition of filterLookup, it is a dictionary of hashset. This is necessary as we are indexing packages with just the package name for the purpose of efficient filtering, and out list could have same packages with different versions.